google/cloud/storage/client.h

```diff
@@ -26,6 +26,9 @@
 
 namespace storage {
 inline namespace STORAGE_CLIENT_NS {
+namespace internal {
```

Comment:
IIUC, generally, we have our package's directory hierarchy mirror the namespace hierarchy -- e.g. everything in the `storage::internal` namespace would be declared and defined under ".../storage/internal", no?  To that effect, is there a reason you can't just include retry_client.h?

...oh. I see, retry_client.h includes this file.  Yay, C++.  Can you leave a comment noting that we need to forward-declare this class because we can't import its header file without creating a dependency cycle?  (Just to answer the "Why is this here?" question I just had, that others will probably have as well.)

Reply:
Done.